lib/util/legacy_dataset_sync.rb

```diff
@@ -29,21 +30,19 @@ def run
 
   # Keep yielding recent packages until the metadata_modified and metadata_created
   # is earlier than yesterday.
-  def get_packages(server)
-    modified_url = "#{server}/api/3/action/package_search?q=metadata_modified:[NOW-1DAY%20TO%20NOW]"
-    created_url = "#{server}/api/3/action/package_search?q=metadata_created:[NOW-1DAY%20TO%20NOW]"
-
-    new_packages = fetch_json(created_url)
-    modified_packages = fetch_json(modified_url)
+  def get_legacy_datasets(server)
+    modified_datasets_url = "#{server}/api/3/action/package_search?q=metadata_modified:[NOW-1DAY%20TO%20NOW]"
+    new_datasets_url = "#{server}/api/3/action/package_search?q=metadata_created:[NOW-1DAY%20TO%20NOW]"
 
-    return unless new_packages || modified_packages
-
-    new_packages['result']['results'].each do |pkg|
-      yield pkg
-    end
+    new_legacy_datasets = fetch_json(new_datasets_url)
+    modified_legacy_datasets = fetch_json(modified_datasets_url)
 
-    modified_packages['result']['results'].each do |pkg|
-      yield pkg
+    [new_legacy_datasets, modified_legacy_datasets].each do |datasets|
+      if there_are? datasets
+        datasets['result']['results'].each do |pkg|
```

Comment:
As @laurentqro mentioned earlier, `each` will do nothing with an empty object, if you want to make sure you do get something when navigating down the structure, use `:fetch` with a default of `{}` instead of `:[]`.

Also in general - ask for forgiveness, not permission, would be easier to let the `IndexError` or `KeyError` happen and then handle it.